.gitattributes

```diff
@@ -0,0 +1,10 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
+
+# Explicitly declare text files you want to always be normalized and converted
+# to native line endings on checkout.
+*.c text
+*.h text
+
+# Declare files that will always have LF line endings on checkout.
+*.sln text eol=lf
```

Comment:
This is not a good practice -- you need to understand the configuration and create your own (which will be one line), copying the example will not do anything that we want, and will actually cause more issues. Committing it also causes more issues because now the history is even messier.

Reply:
I understand it is not good practice, however we have never encountered this issue before and need to start somewhere. We plan to refine the file as we understand and need time to do so. We are doing our best and taking it step by step. I apologize for making the history messier.

Reply:
Since it is getting late, here is a quick summary of what might be useful:

- Use the `.gitattributes` to ensure that even Windows users accept LF and not CRLF.
- Use the `'.gitattributes` to specify that, going forward, `text eol=lf`` to ensure that even Windows users use the end-of-line character adopted by the other GatorGrader developers, who are primarily using Linux and MacOS.
- Convert all line endings in the Python files that are failing the build on AppVeyor to LF instead of CRLF. (There are Python programs or Linux command-line programs for this task).
- Use this PR and the branch as a "testing ground" for ideas until we can get the builds to pass on AppVeyor. Once all of these builds are passing correctly, @schultzh and @corlettim can "re-start" with a new branch and a PR with commit messages that follow the commit standard.
- We could further test this on the Windows computers in my outer office, if time permits. This would help us to confirm that the configuration for this GitHub repository is working correctly on multiple versions of Windows. As needed, I can also investigate this further with a colleague who develops full-time on Windows.

As a side note, I think that once the `.gitattributes` file is working correctly it will automatically perform the conversion correctly when the checkout occurs on AppVeyor. Is that correct?

Are there other technical points that I have overlooked concerning how to fix this issue? @Michionlion, I'm sure that @corlettim and @schultzh will be able to resolve this issue in the near future. It turns out that they have not yet learned about all of these issues concerning line endings on different operating systems, which is to be expected because we've only had others start to develop on Windows this semester. Both @corlettim and @schultzh are reading articles and learning more so as to ensure that they can use this PR as a testing ground and then soon get a full solution up and running! In fact, until I started to review the following references I myself had forgotten about a lot of the challenges associated with supporting development on Windows, Linux, and Mac. I'm sure that this will prove to be a good learning experience for me and everyone who is working to resolve the issue. We'll do our best! :smiley_cat: 

References:

- https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings
- https://stackoverflow.com/questions/36422107/how-to-convert-crlf-to-lf-on-a-windows-machine-in-python
- https://stackoverflow.com/questions/3569997/how-to-find-out-line-endings-in-a-text-file
- https://github.com/psf/black/issues/258
- https://github.com/psf/black/pull/288

Reply:
Also, quick question @Michionlion: when I use a command like `cat -e invoke.py | more` I _only_ see Unix line-endings in the file. If the file is causing Black to fail on Windows, should I be able to see the Windows CRLF endings instead as a `^M`? When I've had this issue before, I could use the aforementioned `cat` command to see the line endings, helping me to know which files I needed to reformat. Thanks for the help &mdash; it has been a long time since I have had to grapple with this issue!

Reply:
I don't think there are any `CRLF` line endings in the repo itself -- otherwise I think Black would fail even on Linux. Instead, I think that AppVeyor by default checks out code with `core.autocrlf` set to `true`. I haven't actually tested this however, and it is kind of weird how the issues only happen in a few files, and not every file. I definitely could be wrong here though -- actual testing is the only way to figure this out, I think!

Reply:
Thanks for confirming this issue @Michionlion, this is also what I am seeing --- no `CRLF` line endings on Linux and no fail of Black on LInux. So, once we change the configuration file to perform the conversion to `LF`, then the idea is that the build should pass correctly, right?

So, let's continue to use this PR as a testing ground for the builds that run on AppVeyor. @corlettim and @schultzh is there any way that you would like me to help out with this task? Please let us know your plans and if there is any way in which we can assist! Thanks for taking the time to investigate this tricky issue further --- your help is greatly appreciated! :smile: